### PR TITLE
BUG: support TimedeltaIndex serialization in fixed stores (GH9635)

### LINKED ISSUE #9635: Hdf5/PyTables support for TimedeltaIndex
When stored in HDFStore, TimedeltaIndex are converted into Int64Index.

```python
df = pd.DataFrame(np.random.normal(size=(10,5)))
df.index = pd.timedelta_range(start='0s',periods=10,freq='1s')
store = pd.HDFStore('test.h5')
store['df'] = df
df = store['df']
print df.index
```
```
Int64Index([0, 1000000000, 2000000000, 3000000000, 4000000000, 5000000000, 6000000000, 7000000000, 8000000000, 9000000000], dtype='int64')
```
Although the backward conversion is easy (pd.TimedeltaIndex(df.index)) , it's surprising.

```python
pd.show_versions()
```
```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-46-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: fr_FR.UTF-8

pandas: 0.15.2-269-ge3d1b0e
nose: 1.3.1
Cython: 0.20.1post0
numpy: 1.8.2
scipy: 0.13.3
statsmodels: None
IPython: 1.2.1
sphinx: 1.2.2
patsy: None
dateutil: 2.4.1
pytz: 2013b
bottleneck: None
tables: 3.1.1
numexpr: 2.2.2
matplotlib: 1.3.1
openpyxl: None
xlrd: None
xlwt: 0.7.5
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.999
httplib2: 0.8
apiclient: None
rpy2: None
sqlalchemy: 0.9.8
pymysql: None
psycopg2: None

```

I would have been happy to help and add this myself, but I really have no idea were is this hiding. I don't even know if I should have looked toward PyTables instead of here.

Thanks !

### PR DESCRIPTION
**Pull request by JonasAbernot**
closes https://github.com/pydata/pandas/issues/9635

Small change to fix issue 9635. I'm a beginner contributor, feel free to crush it out.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
I updated your comment with "closes" and a link to the issue. Now Github will automatically close that issue when we merge this.

**Comment by TomAugspurger** [comment]
A couple more things.

1. Could you add a test for this? Something that fails on master (without your patch), and works on your branch
2. Add a release note in `doc/source/whatsnew/v0.16.1` with a sentence describing the bug.

Actually, that whatsnew file may not exist yet. Once we start merging pull requests for .16.1 it will be there.

File: pandas/io/tests/test_pytables.py [hunk 1]
```diff
@@ -4687,6 +4687,14 @@ def test_duplicate_column_name(self):
             other = read_hdf(path, 'df')
             tm.assert_frame_equal(df, other)
 
+    def test_preserve_timedeltaindex_type(self):
+        df = DataFrame(np.random.normal(size=(10,5)))
```
Comment (jreback) [inline pandas/io/tests/test_pytables.py]: add the issue number as a comment here.

File: pandas/io/tests/test_pytables.py [hunk 2]
```diff
@@ -4687,6 +4687,14 @@ def test_duplicate_column_name(self):
...
+        df.index = timedelta_range(start='0s',periods=10,freq='1s')
+
+        with ensure_clean_store(self.path) as store:
+            
+            store['df'] = df
```
Comment (jreback) [inline pandas/io/tests/test_pytables.py]: use ``tm.assert_frame_equal(store['df'],df)`` for the comparison

File: pandas/io/tests/test_pytables.py [hunk 3]
```diff
@@ -4687,6 +4687,14 @@ def test_duplicate_column_name(self):
...
+            self.assertEqual(type(store['df'].index), TimedeltaIndex)
 
 def _test_sort(obj):
```
Comment (jreback) [inline pandas/io/tests/test_pytables.py]: I don't recall exactly if the fixed stores preserve some meta data, e.g. ``name/freq``. So pls add the attributes (e.g. you can add them with ``timedelta_range(......, name='foo')`` and see if they are preserved (which the above testing method will catch).

**Comment by jreback** [comment]
@JonasAbernot pretty good!

will need a release note (wait till the end of the week after I push the file up after we release 0.16.0)

**Comment by JonasAbernot** [comment]
Woups, this time I may need your help. Are all these commits normal ? I tried the 
```
git fetch upstream
git rebase upstream/master
```
from the wiki.  
Should I come back ?

**Comment by TomAugspurger** [comment]
[This](https://github.com/pydata/pandas/wiki/Using-Git#fetch-and-then-rebase-interactively-to-squash-reword-fix-otherwise-change-some-commits) is the one you want.

The `--interactive` should open up a window in your editor. [Here's](https://help.github.com/articles/about-git-rebase/) some more info. Basically you'll squash them all down to the first commit. Ping us in [chat](https://gitter.im/pydata/pandas) if you have any trouble.

**Pull request closed without merging**